### PR TITLE
Draft - Binder errors translate to locale

### PR DESCRIPTION
**Pull request by anderson-joyle**
Issue #800.

A few comments:

- To avoid breaking internals, new overloads were created.
- New binding config locale is populated from PowerFxConfig.

### CODE REVIEW

**Comment by BruceHaley** [comment]
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Connectors.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Core.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Interpreter.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.LanguageServerProtocol.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Transport.Attributes.dll**

**Comment by BruceHaley** [comment]
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Connectors.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Core.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Interpreter.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.LanguageServerProtocol.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Transport.Attributes.dll**

File: src/libraries/Microsoft.PowerFx.Core/Binding/BindingConfig.cs [hunk 1]
```diff
@@ -19,10 +20,13 @@ internal class BindingConfig
 
         public bool UseThisRecordForRuleScope { get; }
 
-        public BindingConfig(bool allowsSideEffects = false, bool useThisRecordForRuleScope = false)
+        public CultureInfo CultureInfo { get; }
+
+        public BindingConfig(bool allowsSideEffects = false, bool useThisRecordForRuleScope = false, CultureInfo locale = null)
```
Comment (MikeStall) [inline src/libraries/Microsoft.PowerFx.Core/Binding/BindingConfig.cs:25]: this is potentially breaking. 
May need another overload.

File: src/libraries/Microsoft.PowerFx.Core/Binding/BindingConfig.cs [hunk 2]
```diff
@@ -19,10 +20,13 @@ internal class BindingConfig
 
         public bool UseThisRecordForRuleScope { get; }
 
-        public BindingConfig(bool allowsSideEffects = false, bool useThisRecordForRuleScope = false)
+        public CultureInfo CultureInfo { get; }
```
Comment (MikeStall) [inline src/libraries/Microsoft.PowerFx.Core/Binding/BindingConfig.cs:23]: > CultureInfo CultureInfo { get; } [](http://example.com/codeflow?start=14&length=33)

This is explicitly for error messages? Has no other impact on binding.  (unlike parse modes)

File: src/tests/Microsoft.PowerFx.Core.Tests/BindingEngineTests.cs [hunk 3]
```diff
@@ -143,6 +143,19 @@ public void CheckBindError()
 
             Assert.False(result.IsSuccess);
             AssertContainsError(result, "Error 2-5: Name isn't valid. 'foo' isn't recognized");
+        }
+
+        [Theory]
+        [InlineData("3+foo+2", "Error 2-5: Il nome non è valido. \"foo\" non riconosciuto.", "it-IT")]
+        [InlineData("Foo()", "Error 0-5: 'Foo' est une fonction inconnue ou non prise en charge.", "fr-FR")]
+        [InlineData("AAA", "Error 0-3: O nome não é válido. 'AAA' não é reconhecido.", "pt-BR")]
+        public void CheckBindError2(string expression, string expected, string locale)
```
Comment (MikeStall) [inline src/tests/Microsoft.PowerFx.Core.Tests/BindingEngineTests.cs:152]: >CheckBindError2 [](http://example.com/codeflow?start=20&length=15)

great test!

**Comment by BruceHaley** [comment]
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Connectors.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Core.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Interpreter.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.LanguageServerProtocol.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Transport.Attributes.dll**

**Comment by BruceHaley** [comment]
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Connectors.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Core.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Interpreter.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.LanguageServerProtocol.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Transport.Attributes.dll**

File: src/libraries/Microsoft.PowerFx.Core/Binding/BindingConfig.cs (hunk 2 above)
Comment (anderson-joyle) [reply src/libraries/Microsoft.PowerFx.Core/Binding/BindingConfig.cs]: Removed.

File: src/libraries/Microsoft.PowerFx.Core/Binding/BindingConfig.cs (hunk 1 above)
Comment (anderson-joyle) [reply src/libraries/Microsoft.PowerFx.Core/Binding/BindingConfig.cs]: Removed.

**Comment by BruceHaley** [comment]
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Connectors.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Core.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Interpreter.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.LanguageServerProtocol.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Transport.Attributes.dll**

**Pull request closed without merging**